target_starrocks/sinks.py

```diff
@@ -24,6 +29,113 @@ def get_sqlalchemy_url(self, config: dict) -> str:
+    #    ProgrammingError when checking if a table exists.
+    # 2. The dialect insists on VARCHARS having a length, which is not actually required
+    #    however Meltano SDK by default does not set a length for all VARCHARs.
+    #    so instead we force it.
+    def table_exists(
+        self,
+        full_table_name: str | SQLConnector.FullyQualifiedName
+    ) -> bool:
+        """Determine if the target table already exists.
+
+        Args:
+            full_table_name: the target table name.
+
+        Returns:
+            True if table exists, False if not, None if unsure or undetectable.
+        """
+        _, schema_name, table_name = self.parse_full_table_name(full_table_name)
+
+        try:
+            table_exists_result = sa.inspect(self._engine).has_table(
+                table_name, schema_name
+            )
+        except sa.exc.ProgrammingError:
+            self.logger.info(
+                "ANNOYING TABLE CREATION ERROR IS STILL OCCURING WHEN "
+                "CHECKING IF TABLE EXISTS."
+            )
+            table_exists_result = False
+
+        return table_exists_result
+
+    def create_empty_table(
+        self,
+        full_table_name: str | SQLConnector.FullyQualifiedName,
+        schema: dict,
+        primary_keys: t.Sequence[str] | None = None,  # noqa: ARG002
+        partition_keys: list[str] | None = None,
+        as_temp_table: bool = False,  # noqa: FBT001, FBT002
+    ) -> None:
+        """Create an empty target table.
+
+        Args:
+            full_table_name: the target table name.
+            schema: the JSON schema for the new table.
+            primary_keys: list of key properties.
+            partition_keys: list of partition keys.
+            as_temp_table: True to create a temp table.
+
+        Raises:
+            NotImplementedError: if temp tables are unsupported and as_temp_table=True.
+            RuntimeError: if a variant schema is passed with no properties defined.
+        """
+        if as_temp_table:
+            msg = "Temporary tables are not supported."
+            raise NotImplementedError(msg)
+
+        _ = partition_keys  # Not supported in generic implementation.
+
+        _, schema_name, table_name = self.parse_full_table_name(full_table_name)
+        meta = sa.MetaData(schema=schema_name)
+        columns: list[sa.Column] = []
+
+        try:
+            properties: dict = schema["properties"]
+        except KeyError as e:
+            msg = f"Schema for '{full_table_name}' does not define properties: {schema}"
+            raise RuntimeError(msg) from e
+        for property_name, property_jsonschema in properties.items():
+            columns.append(
+                sa.Column(
+                    property_name,
+                    self.to_sql_type(property_jsonschema),
+                ),
+            )
+
+        _ = sa.Table(table_name, meta, *columns)
+        meta.create_all(self._engine, checkfirst=False)
```

Comment:
This is the key change, I also removed the bits that try to set primary keys as starrocks/iceberg don't support that